KthSmallestElementInASortedMatrix.py

```diff
@@ -0,0 +1,58 @@
+class Solution(object):
+    def kthSmallest(self, matrix, k):
+        """
+        :type matrix: List[List[int]]
+        :type k: int
+        :rtype: int
+        """
+        if not matrix:
+            return 0
+        m = len(matrix)
+        n = len(matrix[0])
+        heap = []
+        for i in range(m):
+            self.heapAdd(heap, (matrix[i][0], i, 0))
+        while k > 1:
+            minItem = self.heapPop(heap)
+            x, y = minItem[1], minItem[2]
+            if y + 1 < n:
+                self.heapAdd(heap, (matrix[x][y + 1], x, y + 1))
+            k -= 1
+        return self.heapPop(heap)[0]
+    
+    def heapAdd(self, heap, item):
+        heap.append(item)
+        self.bubbleUp(heap, len(heap) - 1)
+    
+    def heapPop(self, heap):
+        length = len(heap) - 1
```

Comment:
Why do you need this line?
Can just do
```
heap[0], heap[-1] = heap[-1], heap[0]
```
on the next line

Reply:
sure, will do this.